### ISSUE TITLE
Do not allow duplicate animal records that differ by only the name

### ISSUE THREAD

**Issue by lparsons** [Princeton-LSI-ResearchComputing/tracebase#688]
### Inspiration

Animal records (mice) are defined by the metadata associated with them. If two animals are identical in every way except the name, they are very likely errant duplicates.

### Description

When loading animal records, if an incoming animal matches an existing animal exactly but has a different name, the loader should raise an exception and fail to load the animal.

### Comment

If, at some point in the future, we discover a need to handle replicate animals, a "replicate field" (perhaps "mouse number") should be added to the animal model to allow researchers to explicitly define replicates.

-----

### Assumptions

- List of assumptions that the code will not explicitly address/check
- E.g. We will assume input is correct (explaining why there is no validation)

### Requirements

- [ ] 1. List of numbered conditions to be met for the feature
- [ ] 2. E.g. Every column/row must display a value, i.e. cannot be empty
- [ ] 3. Numbers for reference & checkboxes for progress tracking

### Limitations

- A list of things this work will specifically not do
- E.g. This feature will only handle the most frequent use case X

### Affected Components

A tentative list of anticipated repository items that will be changed, labeled
with "add", "delete", or "change".  One item per line.  (Mostly, this will be a
list of files.)

- change: File path or DB table ...
- add: Environment variable or server setting
- delete: External executable or cron job

#### Interface Change description

Describe changes to usage. *E.g. GUI/command-line changes*

#### Code Change Description

Describe code changes planned for the feature. *(Pseudocode encouraged)*

#### Tests

- [ ] 1. A description of at least one test for each requirement above.
- [ ] 2. E.g. Test for req 2 that there's an exception when display value is ''
- [ ] 3. Numbers for reference & checkboxes for progress tracking

**Comment by mneinast** [comment]
Actually I think there could be instances where different animals have the same metadata.  Here is each column in the Animals sheet of the Study Doc:

- Animal Name
- Age
- Sex
- Genotype
- Weight
- Infusate
- Infusion Rate
- Diet
- Feeding Status
- Treatment
- Study

I'm imagining that users will frequently submit studies where 1-2 groups of animals were given the same infusate.  For example, in a study where 13C-valine was infused comparing two different Genotypes...within a single Genotype, the only unique metadata would be `Weight` (`Infusion Rate` is often in proportion to `Weight`).  We will likely encounter different mice with the same `Weight`.

Right now, does the `Animal Name` field accomplishe the same thing that a `Replicate` field would?

**Comment by lparsons** [comment]
> Right now, does the `Animal Name` field accomplishe the same thing that a `Replicate` field would?

Yes, this issue was described last year which was before the work on an animal and sample database got moving. The main concern was that if people weren't careful in their naming we could end up with two animal records for the same animal. However, as we have moved forward, I'm less concerned about that possibility. I would suggest that this issue can be closed as a "won't fix" and we can revisit the issue in the animal and sample tracking project. Do you agree @mneinast and @hepcat72?

**Comment by hepcat72** [comment]
That sounds reasonable to me @lparsons.  However, let me offer an alternative.  We have no way of tracking potential duplicate animals or assessing the extent of it in existing data.  Instead of a loading-prohibitive error, we could just make it a warning that only the user sees in the validation interface.  The curator wouldn't have to bother with it during a load.

In any case, it would be a low priority issue, not necessary for any of the existing milestones.
